### DISCUSSION TITLE
Crashing nextjs app when comes graphql error?

### DISCUSSION THREAD

**Discussion by nekmart** [urql-graphql/urql#3014] [Q&A] [answered]
Hello, I am facing one problem. First of all please see this image-

![image](https://user-images.githubusercontent.com/111189053/223329229-03e60636-f74d-430d-ad7c-455b29f5201b.png)
 and-
![image](https://user-images.githubusercontent.com/111189053/223329284-cd52da39-a4bc-4db6-b949-ae6b28c87848.png)

Here I try to add a list with section name. My backend throws an error when same name exist. It should show only in message or notification. It should not crashed my app. How can I solve it-

here is my code-

`client.ts`-

```
import { createClient, ssrExchange, dedupExchange, cacheExchange, fetchExchange, errorExchange } from "urql";

const isServerSide = typeof window === "undefined";

const ssrCache = ssrExchange({ isClient: !isServerSide });
const client = createClient({
    url: "http://localhost:3001/ecampus",
    exchanges: [
        dedupExchange, cacheExchange, errorExchange({}), ssrCache, fetchExchange
    ],
    fetchOptions: {
        credentials: "include"
    },
});

export { client, ssrCache };
```

And an the mutations-

 const [{ data, error, fetching }, addSection] = useMutation<AddSectionData>(ADD_SECTION);
 
 And-
 
```
export const ADD_SECTION = gql`
mutation addSection($sectionInput: SectionInput!) {
    addSection(sectionInput: $sectionInput) {
      success
      message
    }
}
`;
```

How can I solve the problem?

**Comment by JoviDeCroock** [comment, marked as answer]
The stack trace sais this is thrown within `@apollo/client` you might be using both urql and apollo 😅 urql does not throw errors for exceptions, that would be orchestrated by you using the library, the [errorExchange just calls onError](https://github.com/urql-graphql/urql/blob/main/packages/core/src/exchanges/map.ts#L32).

Both those screenshots are Apollo Client methods from what I can tell

**Reply by siamahnaf** [reply]
I was wrong!

I have another question @JoviDeCroock -

When I print-

```
  const [{ data, error, fetching }, addSection] = useMutation<AddSectionData>(ADD_SECTION);
  console.log(error?.message);
```
  
  It show me- `[GraphQL] Section name exits!`
  
  Here it should show `Section name exits!`
  
  How can I get only message without `[GraphQL]` flag?

**Reply by JoviDeCroock** [reply]
Was this literally you asking a question from a different account? As your last discussion all of this is in the docs, we are not a helpdesk, reading them gets you very far normally, [error docs](https://formidable.com/open-source/urql/docs/basics/errors/)... - we prefix it with `[GraphQL]` which you can omit yourself as well.

The answer here was mostly triggered because of a genuine issue in the discussion, please don't derail it into something else...

**Reply by siamahnaf** [reply]
Yes, Actually I am new and learning, that's because! Please don't mind.
